functions/test/utils/utils.test.js

```diff
@@ -0,0 +1,227 @@
+const {
+  convertDateToTimestamp,
+  convertReleaseDatesToTimestamps,
+  parseGradlePropertiesForVersion,
+  calculateReleaseState,
+  processLibraryNames,
+  parseCommitTitleFromMessage,
+} = require("../../utils/utils.js");
+const RELEASE_STATES = require("../../utils/releaseStates");
+const {expect} = require("chai");
+const {Timestamp} = require("firebase-admin/firestore");
+const sinon = require("sinon");
+
+describe("convertDateToTimestamp", () => {
+  it("should correctly convert a date string to a Firestore Timestamp", () => {
+    const dateString = "2023-07-19";
+    const result = convertDateToTimestamp(dateString);
+    expect(result).to.be.an.instanceof(Timestamp);
+  });
+
+  it("should throw an error if date string format is invalid", () => {
+    const dateString = "invalid-date-format";
+    const consoleErrorStub = sinon.stub(console, "error");
+    expect(() => convertDateToTimestamp(dateString)).to.throw();
+    consoleErrorStub.restore();
+  });
+});
+
+describe("convertReleaseDatesToTimestamps", () => {
+  it("should correctly convert date strings to Firestore Timestamps", () => {
+    const mockReleases = [
+      {
+        name: "Release 1",
+        codeFreezeDate: "2023-07-19",
+        releaseDate: "2023-08-19",
+      },
+      {
+        name: "Release 2",
+        codeFreezeDate: "2023-09-19",
+        releaseDate: "2023-10-19",
+      },
+    ];
+
+    const result = convertReleaseDatesToTimestamps(mockReleases);
+
+    result.forEach((release, index) => {
+      expect(release).to.have.property("codeFreezeDate");
+      expect(release.codeFreezeDate).to.be.an
+          .instanceof(Timestamp);
+      expect(release).to.have.property("releaseDate");
+      expect(release.releaseDate).to.be.an
+          .instanceof(Timestamp);
+    });
+  });
+
+  it("should not convert if date properties are not present", () => {
+    const mockReleases = [
+      {
+        name: "Release 1",
+      },
+    ];
+
+    const result = convertReleaseDatesToTimestamps(mockReleases);
+
+    expect(result[0]).to.deep.equal(mockReleases[0]);
+  });
+});
+
+
+describe("calculateReleaseState", () => {
+  it("should return SCHEDULED if code freeze is more than 2 days away", () => {
+    const now = new Date();
+    const codeFreeze = new Date(now.getTime() + 3 * 24 * 60 * 60 * 1000);
+    const release = new Date(codeFreeze.getTime() + 5 * 24 * 60 * 60 * 1000);
+    expect(calculateReleaseState(codeFreeze, release, false))
+        .to.equal(RELEASE_STATES.SCHEDULED);
+  });
```

Comment:
Put a space between your assertions and your data (this applies to all the tests below this as well):

```js
describe("calculateReleaseState", () => {
  it("should return SCHEDULED if code freeze is more than 2 days away", () => {
    const now = new Date();
    const codeFreeze = new Date(now.getTime() + 3 * 24 * 60 * 60 * 1000);
    const release = new Date(codeFreeze.getTime() + 5 * 24 * 60 * 60 * 1000);

    expect(calculateReleaseState(codeFreeze, release, false))
        .to.equal(RELEASE_STATES.SCHEDULED);
  });
```

Reply:
Resolved in https://github.com/firebase/firebase-release-dashboard/pull/8/commits/15c9d1d1cd01b3c812e1ca32042cbe083c0a0ce8